### PR TITLE
Works controller no longer needs ControllerResource

### PR DESCRIPTION
**Pull request by jcoyne**
The work controller can inherit from ResourceController to avoid code duplication

### CODE REVIEW

File: app/controllers/concerns/hyrax/works_controller_behavior.rb [hunk 1]
```diff
@@ -18,23 +18,6 @@ module WorksControllerBehavior
...
 
-    class_methods do
-      def cancan_resource_class
-        Hyrax::ControllerResource
-      end
-    end
-
-    def build_change_set(resource)
-      change_set_class.new(resource,
-                           depositor: current_user.user_key,
```
Comment (cjcolvar) [inline app/controllers/concerns/hyrax/works_controller_behavior.rb]: Depositor isn't in the `ResourceController` equivalent.  Should it be?

File: app/controllers/concerns/hyrax/works_controller_behavior.rb [hunk 2]
```diff
@@ -18,23 +18,6 @@ module WorksControllerBehavior
       rescue_from WorkflowAuthorizationException, with: :render_unavailable
     end
 
-    class_methods do
-      def cancan_resource_class
```
Comment (cjcolvar) [inline app/controllers/concerns/hyrax/works_controller_behavior.rb]: This method isn't in `ResourceController`.  Should it be?

Comment (jcoyne) [reply app/controllers/concerns/hyrax/works_controller_behavior.rb]: No, we don't need this anymore.

File: app/controllers/concerns/hyrax/works_controller_behavior.rb (hunk 1 above)
Comment (jcoyne) [reply app/controllers/concerns/hyrax/works_controller_behavior.rb]: Good eye. I've restored this (but it's moved to the private section)

**Pull request merged**